beluga-aws-sdk/src/jobs/mod.rs

```diff
@@ -0,0 +1,306 @@
+use std::collections::HashMap;
+
+use beluga_mqtt::{MqttClient, QoS, Subscriber};
+use chrono::prelude::Utc;
+use futures::prelude::stream::{StreamExt, TryStreamExt};
+
+pub use self::data::JobStatus;
+use self::data::{
+    GetPendingJobsExecutionReq, GetPendingJobsExecutionResp, JobExecution, JobExecutionSummary,
+    StartNextPendingJobExecutionReq, StartNextPendingJobExecutionResp, UpdateJobExecutionReq,
+    UpdateJobExecutionResp,
+};
+use crate::error::Error;
+use crate::Result;
+
+mod data;
+mod datetime;
+
+#[derive(Debug, Clone)]
+pub struct JobsClient {
+    mqtt: MqttClient,
+    subscriber: Subscriber,
+}
+
+impl JobsClient {
+    pub async fn new(mqtt: MqttClient) -> Result<Self> {
+        let subscriber = mqtt
+            .subscribe_many(
+                [
+                    get_accepted(mqtt.thing_name()).as_str(),
+                    get_rejected(mqtt.thing_name()).as_str(),
+                    start_next_accepted(mqtt.thing_name()).as_str(),
+                    start_next_rejected(mqtt.thing_name()).as_str(),
+                ],
+                QoS::AtLeastOnce,
+            )
+            .await?;
+
+        Ok(Self { mqtt, subscriber })
+    }
+
+    pub async fn get(&mut self) -> Result<(Vec<Job>, Vec<Job>)> {
+        self.mqtt
+            .publish(
+                format!("$aws/things/{}/jobs/get", self.mqtt.thing_name()).as_str(),
+                QoS::AtLeastOnce,
+                false,
+                bytes::Bytes::copy_from_slice(&serde_json::to_vec(&GetPendingJobsExecutionReq {
+                    token: format!("get-jobs-{}", self.mqtt.thing_name()),
```

Comment:
Shouldn't we pseudo-randomly generate this token (I guess we can just add a counter to it) so we can be sure we're reading the correct message (since the subscriber now buffers messages from the past).

Reply:
Yeah, agree.